File: triangle.java
import java.lang.Math;

class Point {
    private int x;
    private int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public void setPoint(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getx() {
        return x;
    }

    public int gety() {
        return y;
    }
}

class Triangle {
    private Point a;
    private Point b;
    private Point c;

    public Triangle(Point a, Point b, Point c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    private double distance(Point p1, Point p2) {
        return Math.sqrt(Math.pow(p1.getx() - p2.getx(), 2) + Math.pow(p1.gety() - p2.gety(), 2));
    }

    public double getPerimeterFloat() {
        double ab = distance(a, b);
        double bc = distance(b, c);
        double ca = distance(c, a);
        double perimeter = ab + bc + ca;

        return perimeter;
    }

    public int getPerimeter() {
        return (int) Math.floor(getPerimeterFloat());
    }

    public double getAreaFloat() {
        double ab = distance(a, b);
        double bc = distance(b, c);
        double ca = distance(c, a);
        double half_peri = getPerimeterFloat() / 2;
        double area = Math.sqrt(half_peri * (half_peri - ab) * (half_peri - bc) * (half_peri - ca));

        return area;
    }

    public int getArea() {
        return (int) Math.floor(getAreaFloat());
    }

    public String toString() {
        String string = "Triangle: A(" + a.getx() + ", " + a.gety() + "), B(" + b.getx() + ", " + b.gety() + "), C(" + c.getx() + ", " + c.gety() + ")";
        return string;
    }
}

public class Main {
    public static void main(String[] args) {
        Point a = new Point(1, 1);
        Point b = new Point(1, 2);
        Point c = new Point(2, 1);
        Triangle x = new Triangle(a, b, c);

        System.out.println(x.toString());
        System.out.println(x.getPerimeterFloat() + " " + x.getPerimeter());
        System.out.println(x.getAreaFloat() + " " + x.getArea());
    }
}

File: triangle.py
import math


class Point():
    def __init__(self, x: int, y: int) -> None:
        self.x = x
        self.y = y

    def setPoint(self, x: int, y: int) -> None:
        self.x = x
        self.y = y

    def getx(self) -> int:
        return self.x
    
    def gety(self) -> int:
        return self.y
    
def distance(a: Point, b: Point) -> float:
    return math.sqrt((a.getx() - b.getx())*(a.getx() - b.getx()) + (a.gety() - b.gety())*(a.gety() - b.gety()))

class Triangle():
    def __init__(self, a: Point, b: Point, c: Point) -> None:
        self.a = a
        self.b = b
        self.c = c

    def getPerimeterFloat(self) -> float:
        ab = distance(self.a, self.b)
        bc = distance(self.b, self.c)
        ca = distance(self.c, self.a)
        perimeter = ab + bc + ca

        return perimeter
    
    def getPerimeter(self) -> int:
        return math.floor(self.getPerimeterFloat())
    
    def getAreaFloat(self) -> float:
        ab = distance(self.a, self.b)
        bc = distance(self.b, self.c)
        ca = distance(self.c, self.a)
        half_peri = self.getPerimeterFloat()/2
        area = math.sqrt(half_peri*(half_peri-ab)*(half_peri-bc)*(half_peri-ca))
        
        return area
    
    def getArea(self) -> int:
        return math.floor(self.getAreaFloat())
    
    def toString(self) -> str:
        string = f"Triangle: A({self.a.getx()}, {self.a.gety()}), B({self.b.getx()}, {self.b.gety()}), C({self.c.getx()}, {self.c.gety()})"
        return string


if __name__ == "__main__":
    a, b, c = Point(1, 1), Point(1, 2), Point(2, 1)
    x = Triangle(a, b, c)
    print(x.toString())
    print(x.getPerimeterFloat(), x.getPerimeter())
    print(x.getAreaFloat(), x.getArea())